sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -578,8 +578,9 @@ cdef class Splitter:
         cdef:
             unsigned int split_info_idx
             unsigned int best_split_info_idx = 0
+            unsigned int uint_n_allowed_features = n_allowed_features  # needed as uint for the generated C loop
```

Comment:
Can't we just change the signature of the function instead of introducing a new variable ?

Reply:
Yes, this will work too. I dropped the variable and only changed the signature.
I've chosen the first way, because changing function signatures is often followed by many adaptations in code at several places. But in this case it was only one other function to be considered.